src/main/java/games/strategy/engine/random/PlainRandomSource.java

```diff
@@ -38,10 +30,15 @@ private static long getSeed() {
   }
 
   @Override
-  public synchronized int getRandom(final int max, final String annotation) throws IllegalArgumentException {
-    if (random == null) {
-      random = new MersenneTwister(getSeed());
+  public int getRandom(final int max, final String annotation) {
```

Comment:
Possibly out-of-scope for this PR, but is the annotaion paramter really needed?
The only "real usage" I could find is in the PBEMDiceRoller class.
In case it is too complicated to change the interface structure, I'd appreciate it if this interface could contain a default method overload which accepts the String param but does nothing with it and just calls `getRandom(int)` etc., which can then be overriden by the classes that actually need it, not sure if this works out as expected though.

Reply:
`CryptoRandomSource` also uses the annotation by passing it to its associated `IRemoteRandom`.

I'm not sure adding the default method in the interface is going to make things better.  It will then require `CryptoRandomSource` and `PBEMDiceRoller` to override **four** methods, and they will have to choose a default annotation for the annotation-free overloads.

It seems that _callers_ of these methods always provide an annotation.  It's just that 50% of the _implementations_ don't use them.  So adding methods in the interface that do not take the annotation makes it easy for the caller to do the Wrong Thing and provide no annotation, even if the implementation actually uses it.

If you noticed, `PlainRandomSource` did use the annotation as part of an exception message for a precondition check.  I removed that usage, but maybe I shouldn't have.  Maybe the original author felt it would provide some useful context in an error scenario?  Regardless, I could modify the precondition checks as follows:

```java
checkArgument(max > 0, String.format("max must be > 0 (%s)", annotation));
checkArgument(count > 0, String.format("count must be > 0 (%s)", annotation));
```

I can prototype adding the default interface methods, if you like, so we can see what it looks like.  But I think it's big enough and unrelated to this change to justify a separate PR.

Reply:
Basic usage is fine, no problems with that.